src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -1981,8 +1980,8 @@ private String init(boolean initFull) {
                 } else {
                     // an attempt to retreive both the filemetadatas and datafiles early on, so that
                     // we don't have to do so later (possibly, many more times than necessary):
-                    AuthenticatedUser au = session.getUser() instanceof AuthenticatedUser ? (AuthenticatedUser) session.getUser() : null;
-                    datafileService.findFileMetadataOptimizedExperimental(dataset, workingVersion, au);
+                    //AuthenticatedUser au = session.getUser() instanceof AuthenticatedUser ? (AuthenticatedUser) session.getUser() : null;
+                    //datafileService.findFileMetadataOptimizedExperimental(dataset, workingVersion, au);
```

Comment:
You're removing the only two uses of findFileMetadataOptimizedExperimental - can you then remove it's definition in DataFileServiceBean? (I assume you found that findDeep() is as fast or faster than when using this method? findDeep() definitely looks more maintainable.) I'd also suggest just deleting these lines instead of commenting them out.

Reply:
I did remove these comments now. I did not compare the two methods directly, but judging from the description of the experimental method, it was optimized for making as few db queries as possible. This is exactly what findDeep() does. I also did not notice any problems with findDeep() and all datasets load fast now. The main advantage of findDeep() is that it can also be used in write scenarios (the experimental method was only used in the read-only cases). It can also be used outside the dataset page, like it is now also used for indexing. I am now more certain of the findDeep() method (after using in production for several days now), so I dared to remove the experimental method from the code.